File: eCinema/eCinema.Services/Database/Entities/Promotion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCinema.Services.Database.Entities
{
    public class Promotion
    {
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;
        
        [MaxLength(20)]
        public string? Code { get; set; }
        
        [Column(TypeName = "decimal(5, 2)")]
        public decimal DiscountPercentage { get; set; }
        
        public DateTime StartDate { get; set; }
        
        public DateTime EndDate { get; set; }

        public bool IsDeleted { get; set; } = false;

        public virtual ICollection<Reservation> Reservations { get; set; } = new HashSet<Reservation>();
    }
} 

File: UI/ecinema_mobile/lib/models/promotion.dart
class Promotion {
  final int id;
  final String name;
  final String description;
  final String? code;
  final double discountPercentage;
  final DateTime startDate;
  final DateTime endDate;
  final bool isDeleted;

  Promotion({
    required this.id,
    required this.name,
    required this.description,
    this.code,
    required this.discountPercentage,
    required this.startDate,
    required this.endDate,
    required this.isDeleted,
  });

  factory Promotion.fromJson(Map<String, dynamic> json) {
    return Promotion(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      code: json['code'],
      discountPercentage: double.parse(json['discountPercentage'].toString()),
      startDate: DateTime.parse(json['startDate']),
      endDate: DateTime.parse(json['endDate']),
      isDeleted: json['isDeleted'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'code': code,
      'discountPercentage': discountPercentage,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate.toIso8601String(),
      'isDeleted': isDeleted,
    };
  }
}
